pm/sender_test.go

```diff
@@ -0,0 +1,258 @@
+	for i := 0; i < 100; i++ {
+		ticketParams := defaultTicketParams(t)
+		expectedSessionID := hashToHex(ticketParams.RecipientRandHash)
+		sessions = append(sessions, expectedSessionID)
+
+		go func() {
+			sender.StartSession(recipient, TicketParams{
+				FaceValue:         big.NewInt(0),
+				WinProb:           big.NewInt(0),
+				Seed:              big.NewInt(0),
+				RecipientRandHash: ticketParams.RecipientRandHash,
+			})
+			currentCount := atomic.AddInt32(&sessionCount, 1)
+			if currentCount >= 100 {
+				ch <- struct{}{}
+			}
+		}()
+	}
+	// waiting for all session to be created
+	<-ch
+
+	for _, sessionID := range sessions {
+		_, ok := sender.sessions.Load(sessionID)
+		if !ok {
+			t.Errorf("expected to find sessionID in sender. sessionID: %v", sessionID)
+		}
+	}
+}
+
+func TestCreateTicket_GivenNonexistentSession_ReturnsError(t *testing.T) {
+	sender := defaultNewSender(t)
+
+	_, _, _, err := sender.CreateTicket("foo")
+
+	if err == nil {
+		t.Errorf("expected an error for nonexistent session")
+	}
+	if !strings.Contains(err.Error(), "unknown session") {
+		t.Errorf("expected error to contain 'unknown session' but instead got: %v", err.Error())
+	}
+}
+
+func TestCreateTicket_GivenValidSessionId_UsesSessionParamsInTicket(t *testing.T) {
+	sender := defaultNewSender(t)
+	am := sender.accountManager.(*stubAccountManager)
+	am.signShouldFail = false
+	am.saveSignRequest = true
+	am.signResponse = randBytesOrFatal(42, t)
+	senderAddress := sender.accountManager.Account().Address
+	recipient := randAddressOrFatal(t)
+	recipientRandHash := randHashOrFatal(t)
+	ticketParams := TicketParams{
+		FaceValue:         big.NewInt(1111),
+		WinProb:           big.NewInt(2222),
+		Seed:              big.NewInt(3333),
+		RecipientRandHash: recipientRandHash,
+	}
+	sessionID := sender.StartSession(recipient, ticketParams)
+
+	ticket, actualSeed, actualSig, err := sender.CreateTicket(sessionID)
+
+	if err != nil {
+		t.Errorf("error tryint to create a ticket: %v", err)
+	}
+	if ticket.Sender != senderAddress {
+		t.Errorf("expected ticket sender %v to be %v", ticket.Sender, senderAddress)
+	}
+	if ticket.Recipient != recipient {
+		t.Errorf("expeceted ticket recipient %v to be %v", ticket.Recipient, recipient)
+	}
+	if ticket.RecipientRandHash != recipientRandHash {
+		t.Errorf("expected ticket recipientRandHash %v to be %v", ticket.RecipientRandHash, recipientRandHash)
+	}
+	if ticket.FaceValue != ticketParams.FaceValue {
+		t.Errorf("expected ticket FaceValue %v to be %v", ticket.FaceValue, ticketParams.FaceValue)
+	}
+	if ticket.WinProb != ticketParams.WinProb {
+		t.Errorf("expected ticket WinProb %v to be %v", ticket.WinProb, ticketParams.WinProb)
+	}
+	if ticket.SenderNonce != 1 {
+		t.Errorf("expected ticket SenderNonce %d to be 1", ticket.SenderNonce)
+	}
+	if actualSeed != ticketParams.Seed {
+		t.Errorf("expected actual seed %d to be %d", actualSeed, ticketParams.Seed)
+	}
+	if !reflect.DeepEqual(actualSig, am.signResponse) {
```

Comment:
Thoughts on using [bytes.Equal](https://golang.org/pkg/bytes/#Equal) here for clarity on types being compared?